app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt

```diff
@@ -2,34 +2,72 @@ package ch.epfl.skysync.database
 
 import android.util.Log
 import com.google.firebase.Firebase
+import com.google.firebase.firestore.MemoryCacheSettings
 import com.google.firebase.firestore.firestore
+import com.google.firebase.firestore.firestoreSettings
 import java.lang.Exception
+import java.lang.UnsupportedOperationException
 import kotlin.reflect.KClass
 
-/** Represent a connection to a Firestore database */
-class FirestoreDatabase : Database {
+/**
+ * Represent a connection to a Firestore database
+ *
+ * @param useEmulator If true, it will try to connect to Firestore database of a local emulator
+ */
+class FirestoreDatabase(private val useEmulator: Boolean = false) {
   private val db = Firebase.firestore
   private val TAG = "Firebase"
 
-  override fun <T : Any> add(
+  init {
+    if (useEmulator) {
+      val firestore = Firebase.firestore
+      firestore.useEmulator("10.0.2.2", 5002)
```

Comment:
I'm maybe not enough familiar with the emulator but did you check if it also runs fine on the remote github machine? Since it will also be used for automatic testing

Reply:
Yes I setup the CI to start the emulators (@Prolas correct me if I'm wrong)

Reply:
Yes, the emulator is launched before the androitTests are executed